runner/host/dmesg.go

```diff
@@ -41,6 +42,21 @@ func NewDMesgWithContext(ctx context.Context, cfg DMesgConfig) (*DMesg, error) {
 	if ctx == nil {
 		ctx = context.Background()
 	}
+	if runtime.GOOS == "darwin" {
```

Comment:
I think the flow of logic here could be simplified a bit. In essence, the only difference between darwin and Linux here is the `Command` field on the `runner.ShellConfig`. I would propose we create a `runner.ShellConfig` with the Linux settings. Then, we do a quick `if cfg.OS == "darwin"` check, and if it's true, we set the command to `dmesg` instead of `dmesg -T`.

Then, we can pick back up with the `shell, err := runner.NewShellWithContext(ctx, <the name of the shell config variable created in the previous paragraph> ...`